HighAvailabilityModule.Storage.Client/SQLStorageMembershipClient.cs

```diff
@@ -0,0 +1,428 @@
+        }
+
+        public async Task <(string value, string type)> GetDataEntryAsync(string path, string key)
+        {
+            string value;
+            string type;
+            SqlConnection con = new SqlConnection(this.ConStr);
+            string StoredProcedure = GetDataEntrySpName;
+            SqlCommand comStr = new SqlCommand(StoredProcedure, con);
+            comStr.CommandType = CommandType.StoredProcedure;
+            comStr.CommandTimeout = Convert.ToInt32(Math.Ceiling(this.OperationTimeout.TotalSeconds));
+
+            comStr.Parameters.Add("@dpath", SqlDbType.NVarChar).Value = path;
+            comStr.Parameters.Add("@dkey", SqlDbType.NVarChar).Value = key;
+
+            try
+            {
+                await con.OpenAsync();
+                SqlDataReader ReturnedValue = await comStr.ExecuteReaderAsync();
+                if (ReturnedValue.HasRows)
+                {
+                    ReturnedValue.Read();
+                    value = ReturnedValue[0].ToString();
+                    type = ReturnedValue[1].ToString();
+                    ReturnedValue.Close();
+                }
+                else
+                {
+                    value = string.Empty;
+                    type = string.Empty;
+                }
+                return (value, type);
+            }
+            catch (Exception ex)
+            {
+                Trace.TraceError($"Error occured when getting data entry: {ex.ToString()}");
+                throw new InvalidOperationException($"Error occured when getting data entry: {ex.ToString()}");
+            }
+            finally
+            {
+                con.Close();
+                con.Dispose();
+                comStr.Dispose();
+            }
+        }
+
+        public async Task<Guid> TryGetGuid(string path, string key)
+        {
+            var getDataEntry = await GetDataEntryAsync(path, key);
+            string value = getDataEntry.Item1;
+            string type = getDataEntry.Item2;
+
+            if (type == "System.Guid")
+            {
+                return Guid.Parse(value);
+            }
+            else
+            {
+                throw new InvalidOperationException("Input value is not Guid.");
+            }
+        }
+
+        public async Task<string> TryGetString(string path, string key)
+        {
+            var getDataEntry = await GetDataEntryAsync(path, key);
+            string value = getDataEntry.Item1;
+            string type = getDataEntry.Item2;
+
+            if (type == "System.String")
+            {
+                return value;
+            }
+            else
+            {
+                throw new InvalidOperationException("Input value is not string.");
+            }
+        }
+
+        public async Task<int> TryGetInt(string path, string key)
+        {
+            var getDataEntry = await GetDataEntryAsync(path, key);
+            string value = getDataEntry.Item1;
```

Comment:
With the descriptive name,  we can write `string value = getDataEntry.value` here?